docs/content/query/schema.md

```diff
@@ -0,0 +1,387 @@
+# Schema introspection and alteration
+
+## The schema manager
+
+:::warning
+This is experimental work-in-progress and API will change.
+:::
+
+The schema manager is an object that allows you to introspect and alter your
+database schema.
+
+Once you have a functional bridge instance, you may fetch the *schema manager* instance:
+
+```php
+use MakinaCorpus\QueryBuilder\Bridge\Bridge;
+
+\assert($bridge instanceof Bridge);
+
+$schemaManager = $bridge->getSchemaManager();
+```
+
+`$schemaManager` will be an instance of `MakinaCorpus\QueryBuilder\Schema\SchemaManager`.
+
+## Schema introspection
+
+The schema manager allows reading the current schema:
+
+ - read table data,
+ - read table columns data,
+ - read table foreign key and reverse foreign key data.
+
+Other implementations will come later.
+
+:::warning
+When possible, this API uses the `information_schema` tables, nevertheless each database vendor
+has its own dialect, even when considering the information schema. Please be aware that each
+vendor will apply its own access control over information it gives back when reading those
+catalogs, and result may vary depending upon the current user access rights.
+:::
+
+First, you may want to list databases:
+
+```php
+$databases = $schemaManager->listDatabases();
+// $databases is a string array, each value is a database name.
+```
+
+Then get a list of schemas:
+
+```php
+$schemas = $schemaManager->listSchemas('my_database');
+// $schemas is a string array, each value is a schema name.
+```
+
+:::info
+All schema manager methods take a mandatory `$database` parameter, and an optional
+`$schema` parameter: schemas are namespaces inside a database and are not isolated
+from each other, you can work with the full database at once.
+
+Default value for `$schema` is always `public`, which is the default name for at
+least PostgreSQL when you create a database from scratch.
+:::
+
+:::warning
+MySQL doesn't support schemas, this parameter will always be ignored when working
+with it, and `listSchemas()` will always return a single value which is `public`.
+:::
+
+Then fetch some table list:
+
+```php
+foreach ($schemaManager->listTables('my_database', 'my_schema') as $tableName) {
+    // $tableName is a string
+}
+// $databases is a string array, each value is a database name.
+```
+
+And now, you may also load table information this way:
+
+```php
+$table = $schemaManager->getTable('my_database', 'my_table', 'my_schema');
```

Comment:
It is a bit confusing to have the 'table' argument before the 'schema'. I understand why you did this, but it is weird to read

Reply:
May be an intermediate object like "Schema" could solve this ?

```php
$mySchema = $schemaManager->getTable('my_database', 'my_schema');
$table = $mySchema->getTable('my_table');

Reply:
A schema is more or less a namespace, and tables can have relations across different schemas. Most people in real life simply use a unique schema, which is most of the time the default one (`public` in PostgreSQL). Most people won't use the `$schema` argument. That's why it's after.